scripts/import/dbSNP_v2/update_new_sets.pl

```diff
@@ -179,25 +185,36 @@ sub load_all_variation_sets {
 }
 
 sub update_variation_feature_table {
-  # this function after populating the variation_feature_backup table created by inserting from the original table would then update the variation_set_id column uaing the file from the 
-  # dump_new_variation_sets
+  # this function after populating the variation_feature_backup table created by inserting from the original table would then update the variation_set_id column uaing the file from the recalculate
+  # using the parents and parents set id to update the variation feature table 
   my $dbhvar = shift; 
   my $load_file = shift;
 
 
   my $update_temp_vf = $dbhvar->prepare(q{ UPDATE variation_feature SET variation_set_id = ? 
-                                          WHERE variation_id = ? AND variation_set_id = ''});
+                                          WHERE variation_id = ? });
   
   #my %var_data;
   open FH, "<", "$load_file" or die "Can not open $load_file: $!";
   while (<FH>) {
     chomp;
-    my $var_id = (split)[0];
-    my $var_set_id = (split)[1];
-    use Data::Dumper;
-    print Dumper($var_set_id, $var_id);
+    my @fields = split("\t");
+    my $var_id = $fields[0];
+    my $var_set_id = $fields[1];
+    
+
+    my @sets_array; 
+    # to make sure only unique numbers are in the array 
+    foreach my $x (split(',', $var_set_id)){
+      push @sets_array, $x if !grep{$_ eq $x}@sets_array;
```

Comment:
It should be quicker to use a hash here, something like-
sub unique {
  my %u;
  map { $u{$_} = 1; } @_;
  return keys %u;
}